src/Perl6/Actions.nqp

```diff
@@ -7486,7 +7486,7 @@ class Perl6::Actions is HLL::Actions does STDActions {
                     }
                     $was_lexical := 1;
                 }
-                unless $was_lexical {
+                if $*W.is_lexical_marked_ro($target.name) || !$was_lexical {
```

Comment:
Since we explicitly know the cause of failed binding, I'd rather add a new exception and be more elaborate on the cause.